frameworks/RandomForest/__init__.py

```diff
@@ -8,18 +8,20 @@ def setup(*args, **kwargs):
 
 
 def run(dataset: Dataset, config: TaskConfig):
-    from amlb.datautils import impute
+    from amlb.datautils import impute_array
     from frameworks.shared.caller import run_in_venv
 
-    X_train_enc, X_test_enc = impute(dataset.train.X_enc, dataset.test.X_enc)
+    encode = config.framework_params.get('_encode', True)
+    X_train, X_test = impute_array(dataset.train.X_enc, dataset.test.X_enc) if encode else (dataset.train.X, dataset.test.X)
```

Comment:
I notice that we sometimes impute in `exec.py` (decision tree) and sometimes in `__init__.py` (here). Is there a reason for this distinction? If not we should we consistent.

Reply:
hmm, do we? Is it not done in `exec.py`only when this one is not executed in a subprocess (e.g. constantpredictor).
I agree that we should be consistent, as soon as it does not make things more complex.

Reply:
It looks like you're right. Why is it that we had `DecisionTree` run in the main process, but `RandomForest` in a subprocess?

Reply:
I think I used RandomForest as a test+reference implementation framework when implementing the venv+subprocess with caller/callee logic.

Reply:
RF runs fast, that's one of the reason.
I dont know if it's something you could take into consideration for the workflow, when there are changes in `amlb`: run `constantpredictor` + `RandomForest` to cover both use-cases (with/without subprocess).